firedrake/plot.py

```diff
@@ -0,0 +1,36 @@
+import numpy as np
+
```

Comment:
Given that we're going to return a matplotlib object, we should import matplotlib too.  To make it an optional dependency we can do:

```python
try:
    import matplotlib
except ImportError:
    matplotlib = None
```

And then inside the plot function to:

```python
def plot(...):
    if matplotlib is None:
        raise RuntimeError("Matplotlib not importable, is it installed?")
   ...
```